presto-hive/src/test/java/io/prestosql/plugin/hive/metastore/glue/TestHiveGlueMetastore.java

```diff
@@ -96,4 +112,29 @@ public void testStorePartitionWithStatistics()
     {
         testStorePartitionWithStatistics(STATISTICS_PARTITIONED_TABLE_COLUMNS, BASIC_STATISTICS_1, BASIC_STATISTICS_2, BASIC_STATISTICS_1, EMPTY_TABLE_STATISTICS);
     }
+
+    @Test
+    public void testGetPartitions()
+    {
+        try {
+            createDummyPartitionedTable(tablePartitionFormat, CREATE_TABLE_COLUMNS_PARTITIONED);
+            Optional<List<String>> partitionNames = getMetastoreClient().getPartitionNames(tablePartitionFormat.getSchemaName(), tablePartitionFormat.getTableName());
+            assertTrue(partitionNames.isPresent());
+            assertEquals(partitionNames.get(), Arrays.asList("ds=2016-01-01", "ds=2016-01-02"));
+        }
+        catch (Exception ex) {
+            fail("Failed test testGetPartitions", ex);
+        }
+        finally {
+            dropTable(tablePartitionFormat);
+        }
+    }
+
+    @Override
+    protected ConnectorTableHandle getTableHandle(ConnectorMetadata metadata, SchemaTableName tableName)
```

Comment:
Adding this method may change number of test methods effectively run in `TestHiveGlueMetastore`. Please extract to a separate commit.

Reply:
This change will cause all of the tests that depend on existing tables to fail (they are currently skipped by the method in the super class). What is the reason for this change?

Reply:
I found that the getPartitions tests are getting skipped without this change. What is the best way to test them?

Reply:
It's complicated. Travis invokes `presto-hive-hadoop2/bin/run_hive_tests.sh` which

* starts up HDFS and a Hive metastore in Docker
* runs the `presto-hive/src/test/sql/create-test.sql` Hive script to create test tables
* runs the `AbstractTestHive` tests against this environment

The purpose is to create tables in various ways using Hive, then make sure that Presto can read them. That's where the "existing tables" part comes from -- they already exist when the tests run.

It's fine to leave this out, since there are plenty of existing tests that exercise the metastore partition calls. I verified this by running `TestHiveInMemoryMetastore` with logging on the various `getPartition*` metastore calls.